File: TermiBot.Karma/Models/ChangeRequest.cs
using Flurl.Http;

namespace TermiBot.Karma.Models
{
    public class ChangeRequest
    {
        public string Name { get; }
        public int Amount { get; }
        public string Reason { get; }

        public ChangeRequest(string name, int amount)
        {
            Name = name;
            Amount = amount;
        }

        public ChangeRequest(ChangeRequest request, string reason)
        {
            Name = request.Name;
            Amount = request.Amount;
            Reason = reason;
        }
    }
}

File: karma/src/change_request.rs
pub struct ChangeRequest {
    pub name: String,
    pub amount: i64,
}

impl ChangeRequest {
    pub(crate) fn new(name: &str, amount: i64) -> Self {
        Self {
            name: name.to_string(),
            amount,
        }
    }
}
